File: Array/122.Best_Time_to_Buy_and_Sell_Stock_II/C++/122.Best_Time_to_Buy_and_Sell_Stock_II.cpp
class Solution {
public:
    int maxProfit(vector<int> & prices) 
    {
        //======================================//
        // Input type:                          //
        //   - vector<int> ref. (prices vector) //
        // Return type:                         //
        //   - int (result maximun profit)      //
        //======================================//

        /*Initialize*/
        ///// Length of prices vector /////
        int len_prices = prices.size();

        ///// Record traversal pointer /////
        int record_trav_ptr = 1;

        ///// Result, Record maximun profit /////
        int res_max_profit = 0, record_max_profit = 0;


        /*Greedy-based loop traversal*/
        while (record_trav_ptr < len_prices) //Whole
        {
            while ((record_trav_ptr < len_prices) && (prices[record_trav_ptr] >= prices[(record_trav_ptr - 1)])) //Part
            {
                record_max_profit += (prices[record_trav_ptr] - prices[(record_trav_ptr - 1)]); //Keep updating, accumulating

                (record_trav_ptr++); //Keep updating, traversing

            } //Part

            res_max_profit += record_max_profit; //Keep updating, accumulating

            ///// Check if the current profit needs to be reset or not /////
            if (record_max_profit) { record_max_profit &= 0; }
            else { ; }

            (record_trav_ptr++); //Keep updating, traversing

        } //Whole

        return res_max_profit;
    }
};

File: Array/122.Best_Time_to_Buy_and_Sell_Stock_II/Python/122.Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Length of prices array #####
        len_prices = len(prices)

        ##### Record traversal pointers #####
        record_trav_ptr = 1

        ##### Result, Record maximun profit #####
        res_max_profit, record_max_profit = 0, 0


        #############################
        #Greedy-based loop traversal
        while (record_trav_ptr < len_prices):
            while ((record_trav_ptr < len_prices) and (prices[record_trav_ptr] >= prices[(record_trav_ptr - 1)])):
                record_max_profit += (prices[record_trav_ptr] - prices[(record_trav_ptr - 1)]) #Keep updating/accumulating

                record_trav_ptr += 1 #Keep updating/traversing

            res_max_profit += record_max_profit #Keep updating/accumulating

            ##### Check if the current profit needs to be reset or not #####
            if (record_max_profit):
                record_max_profit &= 0 #Update/Reset
            else:
                pass

            record_trav_ptr += 1 #Keep updating/traversing

        return res_max_profit
